doc/yaml_configuration.md

```diff
@@ -60,210 +60,159 @@ are currently four resolver types:
+in the current directory.
 
-It supports four different styles of values:
+Project packages are different from snapshot dependencies (via
+`resolver`) and extra dependencies (via `extra-deps`) in multiple
+ways, e.g.:
 
-#### Package index
+* Project packages will be built by default with a `stack build`
+  without specific targets. Dependencies will only be built if
+  they are depended upon.
+* Test suites and benchmarks may be run for project packages. They are
+  never run for extra dependencies.
 
-Packages can be stated by a name/version combination, which will be
-looked up in the package index (by default, Hackage). The basic syntax
-for this is:
+__Legacy syntax__ Prior to Stack 1.11, it was possible to specify
+dependencies in your `packages` configuration value as well. This
+support has been removed to simplify the file format. Instead, these
+values should be moved to `extra-deps`. As a concrete example, you
+would convert:
 
 ```yaml
-extra-deps:
-- acme-missiles-0.3
-```
-
-Using this syntax, the most recent Cabal file revision available will
-be used. For more reproducibility of builds, it is recommended to
-state the SHA256 hash of the cabal file contents as well, like this:
+packages:
+- .
+- location:
+    git: https://github.com/bitemyapp/esqueleto.git
+    commit: 08c9b4cdf977d5bcd1baba046a007940c1940758
+  extra-dep: true
+- location:
+    git: https://github.com/yesodweb/wai.git
+    commit: 6bf765e000c6fd14e09ebdea6c4c5b1510ff5376
+    subdirs:
+      - wai-extra
+  extra-dep: true
 
-```yaml
 extra-deps:
-- acme-missiles-0.3@sha256:2ba66a092a32593880a87fb00f3213762d7bca65a687d45965778deb8694c5d1
+  - streaming-commons-0.2.0.0
+  - time-1.9.1
+  - yesod-colonnade-1.3.0.1
+  - yesod-elements-1.1
 ```
 
-Or a specific revision number, with `0` being the original file:
+into
 
 ```yaml
-extra-deps:
-- acme-missiles-0.3@rev:0
-```
-
-Note that specifying via SHA256 is slightly more resilient in that it
-does not rely on correct ordering in the package index, while revision
-number is likely simpler to use. In practice, both should guarantee
-equally reproducible build plans.
-
-#### Local file path
-
-Like `packages`, local file paths can be used in `extra-deps`, and
-will be relative to the directory containing the `stack.yaml` file.
+packages:
+- .
 
-```yaml
 extra-deps:
-- vendor/somelib
+  - streaming-commons-0.2.0.0
+  - time-1.9.1
+  - yesod-colonnade-1.3.0.1
+  - yesod-elements-1.1
+  - git: https://github.com/bitemyapp/esqueleto.git
+    commit: 08c9b4cdf977d5bcd1baba046a007940c1940758
+  - git: https://github.com/yesodweb/wai.git
+    commit: 6bf765e000c6fd14e09ebdea6c4c5b1510ff5376
+    subdirs:
+      - wai-extra
 ```
 
-Note that if a local directory can be parsed as a package identifier,
-Stack will treat it as a package identifier. In other words, if you
-have a local directory named `foo-1.2.3`, instead of:
+And, in fact, the `packages` value could be left off entirely since
+it's using the default value.
```

Comment:
It would be good to have some way to automate this?

Reply:
I'm hesitant about this, since our story for writing out `stack.yaml` files is pretty bad right now given that we can preserve comments.